action.yaml

```
@@ -0,0 +1,50 @@
+name: "setup bazel cache"
+
+inputs:
+  bazelrc-path:
+    description: "Absolute path to write configuration in bazelrc format to."
+    default: "/tmp/nvidia.bazelrc"
+    required: false
+
+outputs:
+  bazelrc-path:
+    description: |
+      Absolute path to a file in bazelrc format containing configuration for the remote cache.
+      This is intended to be included in your project's `.bazelrc` with something like `try-import`.
+      See https://bazel.build/run/bazelrc.
+    value: ${{ steps.setup.outputs.bazelrc-path }}
+  grpc-url:
+    description: "gRPC URL to the bazel cache service"
+    value: ${{ steps.setup.outputs.grpc-url }}
+  https-url:
+    description: "HTTPS URL to the bazel cache service"
+    value: ${{ steps.setup.outputs.https-url }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: set up bazel-cache configuration
```

```suggestion
  steps:
    - name: check 
      if: inputs.error-on-invalid-runner
      run: |
        # pseudo-code
        if runner.label != /linux-amd64-cpu\d{1,3}+/; then
          exit 1
        fi
    - name: set up bazel-cache configuration
```

Since we're deliberately only deploying the cache to our cloud runners, should we add a step that errors if the action is used with a non-cloud runner? We can make the error behavior configurable via an input.

I've attached some psuedo-code as an example above.

---

Sure, just pushed https://github.com/nv-gha-runners/setup-bazel-cache/pull/1/commits/db9382baad576e8e09887c8387ccc387414e50d2 with a possible implementation of that. 

Will comment here once I've tested it.

Consulted https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#runner-context to see what `runner.*` stuff is available.